action/protocol/staking/bucket_validation.go

```diff
@@ -75,15 +76,17 @@ func validateBucketSelfStake(featureCtx protocol.FeatureCtx, csm CandidateStateM
 	return nil
 }
 
-func validateBucketWithEndorsement(esm *EndorsementStateManager, bucket *VoteBucket, height uint64) ReceiptError {
-	status, err := esm.Status(bucket.Index, height)
+func validateBucketWithEndorsement(ctx context.Context, esm *EndorsementStateManager, bucket *VoteBucket, height uint64) ReceiptError {
+	featureCtx := protocol.MustGetFeatureCtx(ctx)
+	status, err := esm.Status(featureCtx, bucket.Index, height)
 	if err != nil {
 		return &handleError{
 			err:           err,
 			failureStatus: iotextypes.ReceiptStatus_ErrUnknown,
 		}
 	}
-	if status != Endorsed && status != UnEndorsing {
+	if (featureCtx.LockEndorsement && status != Endorsed) ||
```

Comment:
what about `featureCtx.LockEndorsement && status != Unendorsing` ?
that is, user send `IntentToRevoke`, past waiting period, but did not send actual `Revoke` yet, is the bucket still considered an active endorsement?